lib/OrdinaryDiffEqFIRK/src/algorithms.jl

```diff
@@ -1,20 +1,69 @@
-# FIRK Methods
+function implicit_rk_docstring(description::String,
+    name::String;
+    references::String = "",
+    extra_keyword_description::String = "",
+    extra_keyword_default::String = "")
+    keyword_default = """
+    chunk_size = Val{0}(),
+    autodiff = true,
+    standardtag = Val{true}(),
+    concrete_jac = nothing,
+    diff_type = Val{:forward},
+    linsolve = nothing,
+    precs = DEFAULT_PRECS,
+    """ * extra_keyword_default
 
-"""
-@article{hairer1999stiff,
-title={Stiff differential equations solved by Radau methods},
-author={Hairer, Ernst and Wanner, Gerhard},
-journal={Journal of Computational and Applied Mathematics},
-volume={111},
-number={1-2},
-pages={93--111},
-year={1999},
-publisher={Elsevier}
-}
+    keyword_default_description = """
+    - `chunk_size`: TBD
+    - `autodiff`: TBD
+    - `standardtag`: TBD
+    - `concrete_jac`: TBD
+    - `diff_type`: TBD
+    - `linsolve`: TBD
+    - `precs`: TBD
+    """ * extra_keyword_description
 
-RadauIIA3: Fully-Implicit Runge-Kutta Method
-An A-B-L stable fully implicit Runge-Kutta method with internal tableau complex basis transform for efficiency.
-"""
+    generic_solver_docstring(
+        description, name, "Fully-Implicit Runge-Kutta Method.", references,
+        keyword_default_description, keyword_default
+    )
+end
+
+hairer1999stiff = """@article{hairer1999stiff,
+    title={Stiff differential equations solved by Radau methods},
+    author={Hairer, Ernst and Wanner, Gerhard},
+    journal={Journal of Computational and Applied Mathematics},
+    volume={111},
+    number={1-2},
+    pages={93--111},
+    year={1999},
+    publisher={Elsevier}}"""
+
+extra_keyword_description = """
+    - `extrapolant`: TBD
+    - `fast_convergence_cutoff`: TBD
+    - `new_W_γdt_cutoff`: TBD
+    - `controller`: TBD
+    - `κ`: TBD
+    - `maxiters`: TBD
+    - `smooth_est`: TBD
```

Comment:
@ChrisRackauckas these need to be filled in.

Reply:
I'm hoping we can remove those soon as we swap to the new nonlinear solvers. I'd leave them undocumented for now so it's not public API.